crates/bevy_math/src/smooth.rs

```diff
@@ -0,0 +1,118 @@
+        smooth_time: f32,
+        delta_time: f32,
+    ) -> (Self, Self)
+    where
+        Self: Sized;
+}
+
+macro_rules! impl_smooth_damp {
+    ($t:ty, $f:ty) => {
+        impl SmoothDamp for $t {
+            fn smooth_damp(
+                from: $t,
+                to: $t,
+                velocity: $t,
+                smooth_time: f32,
+                delta_time: f32,
+            ) -> ($t, $t) {
+                let smooth_time = <$f>::max(smooth_time as $f, 0.0001); // ensure smooth_time is positive and non-zero
+                let delta_time = delta_time as $f;
+
+                // from game programming gems 4, chapter 1.10
+                let omega = 2.0 / smooth_time;
+                let x = omega * delta_time;
+
+                let exp = 1.0 / (1.0 + x + 0.48 * x * x + 0.235 * x * x * x);
+                // let exp = 1.0 / (1.0 + x * (1.0 + x * (0.48 + 0.235 * x))); // TODO: profile me, both in debug & release
+
+                let change = from - to;
+                let temp = (velocity + omega * change) * delta_time;
+
+                (
+                    to + (change + temp) * exp,      // position
+                    (velocity - omega * temp) * exp, // velocity
+                )
+            }
+        }
+    };
+}
+
+impl_smooth_damp! {f32, f32}
+impl_smooth_damp! {f64, f64}
+impl_smooth_damp! {Vec2, f32}
+impl_smooth_damp! {Vec3, f32}
+
+/// Smooths value to a goal using a damped spring limited by a maximum speed.
+pub trait SmoothDampMax {
+    /// Smooths value to a goal using a damped spring limited by a maximum speed.
+    ///
+    /// `smooth_time` is the expected time to reach the target when at maximum velocity.
+    ///
+    /// Returns smoothed value and new velocity.
+    fn smooth_damp_max(
+        from: Self,
+        to: Self,
+        velocity: Self,
+        max_speed: f32,
+        smooth_time: f32,
+        delta_time: f32,
+    ) -> (Self, Self)
+    where
+        Self: Sized;
+}
+
+macro_rules! impl_smooth_damp_max {
+    ($t:ty, $f:ty, $clamp:expr) => {
+        impl SmoothDampMax for $t {
+            fn smooth_damp_max(
+                from: $t,
+                to: $t,
+                velocity: $t,
+                max_speed: f32,
+                smooth_time: f32,
+                delta_time: f32,
+            ) -> ($t, $t) {
+                let max_speed = <$f>::max(max_speed as $f, 0.0001); // ensure max speed is positive and non-zero
+                let smooth_time = <$f>::max(smooth_time as $f, 0.0001); // ensure smooth_time is positive and non-zero
+                let delta_time = delta_time as $f;
+
+                // from game programming gems 4, chapter 1.10
+                let omega = 2.0 / smooth_time;
+                let x = omega * delta_time;
+
+                let exp = 1.0 / (1.0 + x + 0.48 * x * x + 0.235 * x * x * x);
+                // let exp = 1.0 / (1.0 + x * (1.0 + x * (0.48 + 0.235 * x))); // TODO: profile me, both in debug & release
```

Comment:
It's probably worth profiling this and then deciding on which ever one we want. And not leaving it as a `TODO` since this just adds technical debt we don't necessarily want.

Reply:
Edit: after creating a _benchmarking_ suit using `criterion`, on my machine I found the second
```rust
1.0 / (1.0 + x * (1.0 + x * (0.48 + 0.235 * x)))
```
to be consistently faster. 
However, it was only consistently by ~0.8ns

Reply:
Ah, thank you very much! The result was slightly different due to float shenanigans and I hadn't manage to get asm output from rust's playground to check... I'm going ot use the second version!